### ISSUE TITLE
Readme example fails to work

### ISSUE THREAD

**Issue by shved** [AlexRudd/cognito-srp#6]
Yo! Trying to get auth token from cognito, and got the readme example to start with, but suddenly got very frustrating error. (all the cred info is stubbed ofc.)

```
$ go run main.go
panic: unable to covert "username" to big Int

goroutine 1 [running]:
github.com/alexrudd/cognito-srp/v3.hexToBig(0xc00017c0e0, 0xe, 0x0)
	/home/shvedchenko/go/pkg/mod/github.com/alexrudd/cognito-srp/v3@v3.0.0/cognitosrp.go:217 +0x12b
github.com/alexrudd/cognito-srp/v3.(*CognitoSRP).PasswordVerifierChallenge(0xc0000da100, 0xc0005c3d20, 0xbfebaf3f9424e7f6, 0x6aeeb356, 0xa1aec0, 0x0, 0x0, 0xc0004a21e0)
	/home/shvedchenko/go/pkg/mod/github.com/alexrudd/cognito-srp/v3@v3.0.0/cognitosrp.go:144 +0x2f9
main.main()
	/home/shvedchenko/workspace/testCognito/main.go:51 +0x6f0
```
As you can see my api username is somehow passed to a hexToBig function which is not taking a USERNAME key of challengeParams anywhere. 

Here is my go.mod
```
module testCognito

go 1.15

require (
	github.com/alexrudd/cognito-srp/v3 v3.0.0
	github.com/aws/aws-sdk-go-v2 v0.30.0
	github.com/aws/aws-sdk-go-v2/config v0.3.0
	github.com/aws/aws-sdk-go-v2/service/cognitoidentityprovider v0.30.0
)
```
And here is my whole main.go, which is almost the same as your readme example. The only huge difference is that I'm three times casting map[string]*string to map[string]string and back because thats what the funcs expecting for parameters. And that is may be the root of ther problem. Although I've inspected the challengeParameters map and there was the SALT key and SRP_B key, and the values was hex, not my username value. Which is not making me happy anyway. Will do a delve debug session tomorrow, but hoping that there is someting significant I've found and should share with you.

```
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	cognitosrp "github.com/alexrudd/cognito-srp/v3"
	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/config"
	cip "github.com/aws/aws-sdk-go-v2/service/cognitoidentityprovider"
	"github.com/aws/aws-sdk-go-v2/service/cognitoidentityprovider/types"
)

func main() {
	// configure cognito srp
	csrp, err := cognitosrp.NewCognitoSRP("username", "superduperpass", "ap-southeast-2_myPoolId", "client", nil)
	if err != nil {
		log.Fatal(err)
	}

	// configure cognito identity provider
	cfg, _ := config.LoadDefaultConfig(
		config.WithRegion("ap-southeast-2"),
		config.WithCredentialsProvider(aws.AnonymousCredentials{}),
	)
	svc := cip.NewFromConfig(cfg)

	gap := make(map[string]string)
	for k, v := range csrp.GetAuthParams() {
		gap[k] = *v
	}

	// initiate auth
	resp, err := svc.InitiateAuth(context.Background(), &cip.InitiateAuthInput{
		AuthFlow:       types.AuthFlowTypeUserSrpAuth,
		ClientId:       aws.String(csrp.GetClientId()),
		AuthParameters: gap,
	})
	if err != nil {
		log.Fatal(err)
	}

	// respond to password verifier challenge
	chp := make(map[string]*string)
	// log.Fatalf("%+v\n", resp.ChallengeParameters)
	for k, v := range resp.ChallengeParameters {
		chp[k] = &v
	}

	if resp.ChallengeName == types.ChallengeNameTypePasswordVerifier {
		challengeResponses, _ := csrp.PasswordVerifierChallenge(chp, time.Now())
		chrs := make(map[string]string)
		for k, v := range challengeResponses {
			chrs[k] = *v
		}

		resp, err := svc.RespondToAuthChallenge(context.Background(), &cip.RespondToAuthChallengeInput{
			ChallengeName:      types.ChallengeNameTypePasswordVerifier,
			ChallengeResponses: chrs,
			ClientId:           aws.String(csrp.GetClientId()),
		})
		if err != nil {
			log.Fatal(err)
		}

		// print the tokens
		fmt.Printf("Access Token: %s\n", *resp.AuthenticationResult.AccessToken)
		fmt.Printf("ID Token: %s\n", *resp.AuthenticationResult.IdToken)
		fmt.Printf("Refresh Token: %s\n", *resp.AuthenticationResult.RefreshToken)
	} else {
		log.Fatal("hui")
	}
}
```

**Comment by shved** [comment]
BTW thanks for the work you do! All the AWS API is so frustrating, and you're trying to make it better to work with :heart:

**Comment by alexrudd** [comment]
Oh my god - they've migrated back to `map[string]string` from `map[string]*string`. I literally just updated the package to better support the pointer maps 😭 

I think what's happening in your case is in the conversion from `map[string]string` to `map[string]*string` you're referencing a loop scoped variable that has its value changed every iteration:

```go
for k, v := range resp.ChallengeParameters {
    chp[k] = &v
}
```

should be:

```go
for k, v := range resp.ChallengeParameters {
    scopedV := v
    chp[k] = &scopedV
}
```

The easiest fix will be to use `v2` of this repo, as that uses `map[string]string` throughout. In your above example just replace `github.com/alexrudd/cognito-srp/v3` with `github.com/alexrudd/cognito-srp/v2`.

I'll release a new version at the weekend so we're back to `map[string]string`

**Comment by shved** [comment]
Thanks alex! So shame I've fallen into this common mistake with iterations. 

BTW dont clue a single reason for them to use string pointers instead of strings. Thanks for help again.

**Comment by alexrudd** [comment]
Got a chance to fix this. Released `v4.0.0` which can be imported using `cognitosrp "github.com/alexrudd/cognito-srp/v4"`

Thanks for opening the issue :)
